include/partition/edge_based_graph_reader.hpp

```diff
@@ -0,0 +1,202 @@
+#ifndef OSRM_EDGE_BASED_GRAPH_READER_HPP
+#define OSRM_EDGE_BASED_GRAPH_READER_HPP
+
+#include "storage/io.hpp"
+#include "util/coordinate.hpp"
+#include "util/dynamic_graph.hpp"
+#include "util/typedefs.hpp"
+
+#include <cstdint>
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <vector>
+
+namespace osrm
+{
+namespace partition
+{
+
+struct EdgeBasedGraphEdgeData
```

Comment:
Well I started out only exposing _some_ member attributes. Then had to add some for the edge based graph preparation step. Then added the boundary arc flag (which we're currently not using). And finally added all extractor edge members in addition, since we have to serialize out the full edge based graph after modification anyway, right?